### DISCUSSION TITLE
Apollo-sv contains many classes that should be in other ontologies

### DISCUSSION THREAD

**Discussion by cmungall** [ApolloDev/apollo-sv#277] [Ontology interoperability harmonization and orthogonality]
apollo-sv contains many classes that are out of scope and should be in other OBO ontologies. Note that in many cases here the deficiency is in OBO, either existing vocabularies do not accept terms or we have not agreed OBO-wide where certain terms belong. This problem is perpetuated by the use of BFO shadow classes (e.g. foo + foo datum). While not necessarily apollo-sv's problem, I would recommend this team is proactive in making tickets on other ontology repos to get terms submitted and reporting back to OBO where there are problems.

In OBO we also need a better way of indicating which ontologies typically mint classes that may eventually be housed elsewhere.

Here is a random sampling:

we still don't have a good story for symbiosis types in OBO cc @diatomsRcool, but these should probably be housed elsewhere:

id|label
---|---
APOLLO_SV:00000337|ectosymbiosis
APOLLO_SV:00000338|antagonism
APOLLO_SV:00000339|amensalism
APOLLO_SV:00000340|agonism
APOLLO_SV:00000341|intimate agonism
APOLLO_SV:00000342|ectosymbiotic parasitism
APOLLO_SV:00000343|facultative ectosymbiotic parasitism
APOLLO_SV:00000344|obligatory ectosymbiotic parasitism
APOLLO_SV:00000345|ectosymbiotic parasitoidism
APOLLO_SV:00000346|facultative ectosymbiotic parasitoidism
APOLLO_SV:00000347|obligatory ectosymbiotic parasitoidism
APOLLO_SV:00000348|ectosymbiotic commensalism
APOLLO_SV:00000349|facultative ectosymbiotic commensalism
APOLLO_SV:00000350|obligatory ectosymbiotic commensalism
APOLLO_SV:00000351|ectosymbiotic mutualism
APOLLO_SV:00000352|facultative ectosymbiotic mutualism
APOLLO_SV:00000353|obligatory ectosymbiotic mutualism
APOLLO_SV:00000354|endosymbiosis
APOLLO_SV:00000355|endosymbiotic parasitism
APOLLO_SV:00000356|transient agonism
APOLLO_SV:00000357|predation
APOLLO_SV:00000358|grazing
APOLLO_SV:00000359|facultative endosymbiotic parasitism
APOLLO_SV:00000360|obligatory endosymbiotic parasitism
APOLLO_SV:00000361|endosymbiotic parasitoidism
APOLLO_SV:00000362|facultative endosymbiotic parasitoidism
APOLLO_SV:00000363|obligatory endosymbiotic parasitoidism
APOLLO_SV:00000364|endosymbiotic commensalism
APOLLO_SV:00000365|facultative endosymbiotic commensalism
APOLLO_SV:00000366|obligatory endosymbiotic commensalism
APOLLO_SV:00000367|endosymbiotic mutualism
APOLLO_SV:00000368|facultative endosymbiotic mutualism
APOLLO_SV:00000369|obligatory endosymbiotic mutualism
APOLLO_SV:00000596|symbiosis

ENVO or PCO cc @pbuttigieg @ramonawalls

id|label
---|---
APOLLO_SV:00000097|ecosystem
APOLLO_SV:00000094|community
APOLLO_SV:00000113|atmosphere of the Earth
APOLLO_SV:00000035|organism population universal

I believe that environmental variables are currently scoped for ENVO but this could be discussed

id|label
---|---
APOLLO_SV:00000513|mean rate of hydrological precipitation
APOLLO_SV:00000514|mean air temperature

STATO or IAO (or EDAM, which is not OBO):

id|label
---|---
APOLLO_SV:00000076|μ
APOLLO_SV:00000077|σ
APOLLO_SV:00000078|beta distribution
APOLLO_SV:00000079|deprecated log normal distribution
APOLLO_SV:00000080|uniform distribution
APOLLO_SV:00000081|triangular distribution
APOLLO_SV:00000082|Bayesian network
APOLLO_SV:00000434|statistical confidence

Even though it's not OBO, I would use EDAM here. Or we can raise this to an OBO level ticket:

id|label
---|---
APOLLO_SV:00000203|NCBI taxon identifier
APOLLO_SV:00000319|NLM ID
APOLLO_SV:00000320|ISSN Linking
APOLLO_SV:00000321|ISO Title Abbreviation
APOLLO_SV:00000324|NLM Title Abbreviation
APOLLO_SV:00000325|URI creation date

diseases

id|label
---|---
APOLLO_SV:00000574|acute respiratory illness

OBA cc @matentzn

id|label
---|---
APOLLO_SV:00000471|minimum eggs per oviposition site
APOLLO_SV:00000472|age-dependent fecundity reduction
APOLLO_SV:00000442|migration speed

**Comment by ramonawalls** [comment]
I don't know what a "organism population universal" is, but f you can define it, I'm happy to include it in PCO, assuming it does not already exist with another label.

**Comment by hoganwr** [comment]
Plenty of stories behind many of these terms that perhaps could inform how
to prevent the problem from arising again.  Would be happy to share those
in whatever is an appropriate forum.

We tried working with ENVO.  Sometimes when you need a term, you need it
pretty quickly.  Delay on the part of the host "ontology" leaves you with
the idea of "well, we'll stick it here now and fix it later when the proper
ontology finally adds it."  I'm guilty of that as an owner of multiple
ontologies.

Many of these are annotation properties, I think (NLM Title Abbreviation?)

Never heard of OBA and would guess we created those terms  before it
existed.

The symbiosis terms were a student project presented at ICBO and published
in open access journal.  Was a very systematic approach.  How to handle
student projects?  Hold up PhDs waiting for terms?

**Comment by hoganwr** [comment]
Also, Apollo-SV long predated STATO

**Comment by hoganwr** [comment]
Also, we recognize the issue and just ceded about 10 terms back to IAO,
which by the way, was a pretty dead letter when we created those classes.

**Comment by diatomsRcool** [comment]
and I am guilty of not being fast enough and of creating out-of-scope terms :(
I will add that I have several symbiosis-related terms in ecocore (which is a new ontology), so those APOLLO terms could be re-homed there.

**Comment by dillerm** [comment]
We published a [paper](https://bmcmedinformdecismak.biomedcentral.com/articles/10.1186/s12911-020-01273-0) on those symbiosis classes explaining the need to create these classes, and it basically boiled down to there not being suitable or detailed representations of these interactions (note: ECOCORE, to my knowledge, did not exist when we started working on these classes). We did have [some conversations](https://github.com/geneontology/go-ontology/issues/16909) with GO about 'symbiosis' since they also have a class for it, but I think we decided that their approach of constraining it to being only about the molecular/genetic aspects of such interactions was too restricting (they originally had a definition that was really good and then changed it).

**Comment by cmungall** [comment]
Thanks all for your comments - it seems apollo-sv has been diligent in
trying to get the appropriate terms in different ontologies (including the
exemplary practice of filing tickets in github repos rather than emails).
It seems that something is not working right at the OBO level, it should
not be this hard to get terms in. There are some genuinely hard problems
such as how best to modularize a complex domain like ecology / symbiotic
interactions. Maybe some kind of dedicated meeting  (virtual) around this
is required perhaps around ICBO time?

>

**Comment by cmungall** [comment]
Do you happen to have the ENVO tickets handy?

**Comment by dillerm** [comment]
Here is the ENVO ticket for the terms you listed: https://github.com/EnvironmentOntology/envo/issues/387.

**Comment by zhengj2007** [comment]
'ORCID identifier' was add in IAO with ID, IAO_0000708.

[APOLLO_SV_00000496](http://purl.obolibrary.org/obo/APOLLO_SV_00000496) (ORCID) should be deprecated and add annotation 'term replaced by': http://purl.obolibrary.org/obo/IAO_0000708

**Comment by hoganwr** [comment]
The IAO purl still points to a 2020 version, so how to update Apollo-SV before getting my class back?

**Comment by zhengj2007** [comment]
I am preparing the new release of IAO now and will let you know when it's out. @hoganwr

**Comment by hoganwr** [comment]
Great - thanks!

**Comment by StroemPhi** [comment]
@hoganwr the IAO PURL now points to the new release, so APOLLO-SV:ORCID can be obsoleted.

**Comment by hoganwr** [comment]
Thanks!  I've done that, made a new release of the ontology (w/a slight reorganization), and updated the PURL to always point to the latest.  Once the PURL redirects, hopefully ontobee picks it up and pulls in the new version, and then all will be well.
